crates/ironrdp-server/src/server.rs

```diff
@@ -0,0 +1,224 @@
+use std::{io::Cursor, net::SocketAddr};
+
+use anyhow::{Error, Result};
+use bytes::BytesMut;
+use ironrdp_pdu::{
+    self,
+    input::{
+        fast_path::{FastPathInput, FastPathInputEvent},
+        InputEventPdu,
+    },
+    mcs, rdp, Action, PduParsing,
+};
+use ironrdp_tokio::{Framed, FramedRead, FramedWrite, TokioFramed};
+use tokio::{net::TcpListener, select};
+use tokio_rustls::TlsAcceptor;
+
+use crate::{
+    acceptor::{self, BeginResult, ServerAcceptor},
+    builder, capabilities,
+    display::{DisplayUpdate, RdpServerDisplay},
+    encoder::UpdateEncoder,
+    handler::RdpServerInputHandler,
+};
+
+pub struct RdpServer<H, D> {
+    opts: RdpServerOptions,
+    handler: H,
+    display: D,
+}
```

Comment:
I usually try to avoid dynamic dispatch if possible, especially since there should only be a single server instance in the codebase. But you're right, I haven't considered compilation time, and performance shouldn't be an issue here.

I agree, I'll box them.

Reply:
Thank you! IMO your point is absolutely valid as well: there will be only one monomorphized version, so at least there is no code size issue. But when it ends up in downstream code I prefer to be more careful… I would use static dispatch without too much hesitation if it’s an implementation detail (all monomorphized code is directly inside the same codegen unit), and of course for "small and shallow" generic code.
I take inspiration from bevy: it relies heavily on dynamic dispatching to achieve its simple(r) API and amazing iteration cycle speed
Keeping all this in mind, don’t hesitate to challenge this approach when you think it’s not appropriate

Reply:
Yeah, I never thought about the codegen unit issue until you pointed it out, but now it makes sens. 
Thanks for the tips! I haven't really worked on a lib before, this feedback is very useful to me :)

Reply:
Done